jaeger-core/src/main/java/com/uber/jaeger/Tracer.java

```diff
@@ -197,11 +202,31 @@ public void close() {
     @Override
     public io.opentracing.Tracer.SpanBuilder addReference(
         String referenceType, io.opentracing.SpanContext referencedContext) {
-      if (referencedContext instanceof SpanContext
-          && (Utils.equals(referenceType, References.CHILD_OF)
-                      || Utils.equals(referenceType, References .FOLLOWS_FROM))) {
-        this.references.add(new Reference((SpanContext) referencedContext, referenceType));
-        this.baggage.putAll(((SpanContext)referencedContext).baggage());
+
+      if (!(referencedContext instanceof SpanContext)) {
+        return this;
+      }
+
+      // TODO UI currently does not support arbitrary references
+      if (Utils.equals(referenceType, References.CHILD_OF)
+          || Utils.equals(referenceType, References .FOLLOWS_FROM)) {
```

Comment:
nit: could negate the condition and `return this`, to decrease the nesting of the main code